src/qt/macdockiconhandler.mm

```diff
@@ -44,6 +46,25 @@ - (void)handleDockClickEvent:(NSAppleEventDescriptor*)event withReplyEvent:(NSAp
 
 @end
 
+MacDockShutdownHandler::MacDockShutdownHandler() {
+    Class delClass = (Class)[[[NSApplication sharedApplication] delegate] class];
+    SEL closeHandle = sel_registerName("applicationShouldTerminate:");
+    if (class_getInstanceMethod(delClass, closeHandle)) {
+        class_replaceMethod(delClass, closeHandle, (IMP)MacDockShutdownHandler::handleShutdown, "B@:");
+    } else {
+        class_addMethod(delClass, closeHandle, (IMP)MacDockShutdownHandler::handleShutdown,"B@:");
+    }
+}
+
+MacDockShutdownHandler::~MacDockShutdownHandler() {}
```

Comment:
This one can be nuked, unless it's ported from Bitcoin. If so, keep it to stay in line with upstream :)

Reply:
I was planning on de-hooking but the application is already exiting, so why bother? Nuked.